resources/dw/cluster/aws/resource_cluster.go

```diff
@@ -133,4 +164,51 @@ func (r *dwClusterResource) Delete(ctx context.Context, req resource.DeleteReque
 		)
 		return
 	}
+
+	if state.PollingOptions.Async.ValueBool() {
+		callFailedCount := 0
+		stateConf := &retry.StateChangeConf{
+			Pending:      []string{"Deleting", "Running"},
+			Target:       []string{"Deleted"}, // This is not an actual state, we added it to fake the state change
+			Delay:        30 * time.Second,
+			Timeout:      state.getPollingTimeout(),
+			PollInterval: 30 * time.Second,
+			Refresh:      r.stateRefresh(ctx, clusterID, &callFailedCount, int(state.PollingOptions.CallFailureThreshold.ValueInt64())),
+		}
+		_, err := stateConf.WaitForStateContext(ctx)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error waiting for data warehouse aws cluster",
+				"Could not delete cluster, unexpected error: "+err.Error(),
+			)
+			return
+		}
+	}
+}
+
+func (r *dwClusterResource) stateRefresh(ctx context.Context, clusterID *string, callFailedCount *int, callFailureThreshold int) func() (any, string, error) {
+	return func() (any, string, error) {
+		tflog.Debug(ctx, "About to describe cluster")
+		params := operations.NewDescribeClusterParamsWithContext(ctx).
+			WithInput(&models.DescribeClusterRequest{ClusterID: clusterID})
+		resp, err := r.client.Dw.Operations.DescribeCluster(params)
+		if err != nil {
+			if strings.Contains(err.Error(), "NOT_FOUND") {
+				return &models.DescribeClusterResponse{}, "Deleted", nil
+			}
+			*callFailedCount++
+			if *callFailedCount <= callFailureThreshold {
+				tflog.Warn(ctx, fmt.Sprintf("Error describing data warehouse aws cluster with call failure "+
+					"due to [%s] but threshold limit is not reached yet (%d out of %d).", err.Error(), callFailedCount, callFailureThreshold))
+				return nil, "", nil
+			}
+			tflog.Error(ctx, fmt.Sprintf("Error describing data warehouse aws cluster (due to: %s) and call "+
+				"failure threshold limit exceeded.", err))
+			return nil, "", err
+		}
+		*callFailedCount = 0
+		cluster := resp.GetPayload()
+		log.Printf("Described cluster: %s", cluster.Cluster.Status)
```

Comment:
Is this going to the Stdout? use tflog insted
```suggestion
		tflog.Info("Described cluster: %s", cluster.Cluster.Status)
```